### PR TITLE
BEMHTML: Omit optional closing tags (fix for #360)

### PR DESCRIPTION
**Pull request by miripiruni**
Instead of https://github.com/bem/bem-xjst/pull/215

### CODE REVIEW

**Comment by miripiruni** [comment]
```
Test started…
{ _: [],
  h: false,
  help: false,
  rev1: '3fa75a30087a4339dd3b6cde1f661d8a281155bc',
  rev2: '5ab2e2fc3d13ef95ae6c1fdd34f3cfe9f25580db',
  bemjson: 10000,
  dataPath: '/Users/miripiruni/bemjson',
  templatePath: './node_modules/web-data/templates.js',
  '$0': 'runner.js',
  repeat: undefined,
  verbose: undefined }
Tue Oct 04 2016 13:04:24 GMT+0300 (MSK)
Total test time:  269662.665192
./lib/compare.py ./dat-3fa75-5ab2e ./dat-3fa75-5ab2e/3fa75a30087a4339dd3b6cde1f661d8a281155bc-10000-1475575599337.dat ./dat-3fa75-5ab2e/5ab2e2fc3d13ef95ae6c1fdd34f3cfe9f25580db-10000-1475575733711.dat
Percentile:  0.5
{ rev1: 8.781342,
  rev2: 8.879855,
  'diff abs': 0.09851299999999874,
  'diff percent': 1.1093987458128396 }

Percentile:  0.9
{ rev1: 13.329163,
  rev2: 13.328021,
  'diff abs': 0.0011419999999997543,
  'diff percent': 0.008567679756033009 }

Percentile:  0.95
{ rev1: 15.027906,
  rev2: 15.050939,
  'diff abs': 0.02303299999999986,
  'diff percent': 0.15303364128975705 }
```

screen shot 2016-10-04 at 13 25 27

**Comment by miripiruni** [comment]
cc @vithar

**Comment by vithar** [comment]
👍

**Comment by zxqfox** [comment]
👍

File: docs/ru/3-api.md [hunk 1]
```diff
@@ -209,6 +210,45 @@ var html = templates.apply(bemjson);
...
+    }
+};
+
+var html = templates.apply(bemjson);
+```
+
+В результате `html` будет содержать строку:
+
+```html
+<table><tr><th>table header<td>table cell
```
Comment (miripiruni) [inline docs/ru/3-api.md]: Опечатка

File: lib/bemhtml/index.js [hunk 2]
```diff
@@ -152,14 +162,15 @@ BEMHTML.prototype.renderClose = function renderClose(prefix,
     if (content || content === 0)
       out += this.renderContent(content, isBEM);
 
-    out += '</' + tag + '>';
+    if (!this._omitOptionalEndTags ||
+        (this._omitOptionalEndTags && !OPTIONAL_END_TAGS.hasOwnProperty(tag)))
```
Comment (miripiruni) [inline lib/bemhtml/index.js]: Можно упростить.

**Pull request merged**